Updates for IETF 105 (#11)
IPv6-mPvD/mpvd-ietf-drafts: draft-ietf-intarea-provisioning-domains.md

---
title: Discovering Provisioning Domain Names and Data
abbrev: Provisioning Domains

> docname: draft-ietf-intarea-provisioning-domains-06

Reviewer: is this needed?

Author: Reverted!

--- abstract

Provisioning Domains (PvDs) are defined as consistent
sets of network configuration information. This allows hosts to manage

> connections to multiple networks and interfaces simulataneously, such as

Reviewer: typo simulataneously, run idspell?

Author: Thanks for catching!

when a home router provides connectivity through both a broadband and
cellular network provider.

This document defines a mechanism for explicitly identifying PvDs through
a Router Advertisement (RA) option. This RA option announces a PvD identifier,
which hosts can compare to differentiate between PvDs. The option can directly

> carry some information about a PvD, can optionally point to additional

Reviewer 2: ```suggestion
carry some information about a PvD and can optionally point to additional
```

PvD information that can be retrieved using HTTP over TLS.

--- middle

# Introduction

Provisioning Domains (PvDs) are defined in {{?RFC7556}} as consistent
sets of network configuration information. This information includes
properties that are traditionally associated with a single networking
interface, such as source addresses, DNS configuration, proxy configuration,

> and gateways addresses.

Reviewer 2: ```suggestion
and gateway addresses.
```

internet, multiple PvDs (with different PvD IDs) can be provisioned on
a single host interface. Similarly, the same PvD ID could be used on
different interfaces of a host in order to inform that those PvDs
ultimately provide equivalent services.

This document also introduces a mechanism for hosts to retrieve optional
additional information related to a specific PvD by means of an HTTP
over TLS query using an URI derived from the PvD ID. The retrieved JSON
object contains additional information that would typically be
considered too large to be directly included in the Router
Advertisement, but might be considered useful to the applications, or
even sometimes users, when choosing which PvD should be used.

For example, if Alice has both a cellular network provider and a
broadband provider in her home, her PvD-aware devices and applications
would be aware of both available uplinks via her local Wi-Fi. These applications
could fail-over between these networks, or run connections over both
(potentially using multi-path transports). Applications could also select
specific uplinks based on the properties of the network; for example,

> the cellular network provides free high-quality video streaming,

Reviewer 2: ```suggestion
if the cellular network provides free high-quality video streaming,
```

compression described in Section 4.1.4 of {{!RFC1035}} MUST NOT be used.

Padding:

> : Zero or more padding octets to the next 8 octets boundary (see Section 4.6 of

Reviewer 2: ```suggestion
: Zero or more padding octets to the next 8 octet boundary (see Section 4.6 of
```

: Zero or more RA options that would
otherwise be valid as part of the Router Advertisement main body,
but are instead included in the PvD Option such as to be ignored
by hosts that are not PvD-aware.

Here is an example of a PvD option with "example.org" as the

> PvD ID FQDN and including a RDNSS and prefix information options.

Reviewer 2: ```suggestion
PvD ID FQDN and including RDNSS and prefix information options.
```
or 
```suggestion
PvD ID FQDN and including a RDNSS and prefix information option.
```

scope of this document. Useful insights about these considerations can
be found in {{?I-D.kline-mif-mpvd-api-reqs}}.

### DHCPv6 configuration association {#dhcpv6}

DHCPv6 {{?RFC8415}} configuration elements can be associated
with Implicit or Explicit PvDs. If a DHCPv6 message contains
an assignment of IPv6 addresses or prefixes, these addresses
or prefixes can fall within the Prefix Information Option (PIO)
defined by an RA. Information carried in such DHCPv6 messages
MUST be associated with the PvD (Explicit or Implicit) defined
by the matching RA. RAs that do not contain a PIO cannot be
associated with DHCPv6.

Configuration elements retrieved via DHCPv6 that are associated
with IPv6 addresses or prefixes that do not match any PIO sent in
an RA MUST NOT be associated with any RA-based PvD. Hosts
MAY treat this information as a separate Implicit PvD.

Stateless configuration elements (elements not associated with an
IPv6 address or prefix) MUST be associated with all Implicit PvDs
on the same interface.

### DHCPv4 configuration association {#dhcpv4}

Associating DHCPv4 {{?RFC2131}} configuration elements with Explicit PvDs allows

> hosts to treat a set of IPv4 and IPv6 configuration as a single PvD

Reviewer 2: ```suggestion
hosts to treat a set of IPv4 and IPv6 configurations as a single PvD
```

with shared properties. For example, consider a router that provides two different
uplinks. One could be a broadband network that has data rate and streaming

> properties described in PvD additional information, and provides both IPv4

Reviewer 2: ```suggestion
properties described in PvD additional information and that provides both IPv4
```

prefixes (under the key "prefixes") which MUST be checked against all
the Prefix Information Options advertised in the RA. If any of the
prefixes included in the PIO is not covered by at least one of the
listed prefixes, the associated PvD information MUST be considered

> to be misconfigured, and MUST NOT be used by the host. See

Reviewer 2: `to be misconfigured` or `to be a misconfiguration`?
In case the latter: 
```suggestion
to be a misconfiguration, and MUST NOT be used by the host. See
```

example.org., R-flag = 0 (actual length of the header with padding
24 bytes = 3 * 8 bytes)

> 	- Recursive DNS Server: length = 5, addresses= \[2001:db8:cafe::53, 2001:db8:f00d::53\]

Reviewer: bad whitespace?

Author: Fixed!

It must be noted that {{misconfig}} of this document
only provides reasonable assurance against misconfiguration but does not
prevent an hostile network access provider to advertize wrong
information that could lead applications or hosts to select an hostile PvD.

Users cannot be assumed to be able to meaningfully differentiate between
"safe" and "unsafe" networks. This is a known attack surface that is present
whether or not PvDs are in use, and hence cannot be addressed by this document.
However, a host that correctly implements the MPvD architecture ({{?RFC7556}})
using the mechanism described in this document will be less susceptible to such

> attacks than a host that does not.

Reviewer 2: Do we need to say why this is the case?

Author: Added a bit of text here

Reviewer 2: 👍 Looks good!
